### ISSUE TITLE
[bug]: watchtowers: need to correctly handle acked updates for closed channels

### ISSUE THREAD

**Issue by ellemouton** [lightningnetwork/lnd#7980]
### Background

The logs shown in #7959 reveal a small watchtower client bug: 
Basically if we have a updates queued in the tower client already but then the channel is closed, there is a chance that all the info for the channel could be deleted from the tower client and then when the ACK for the update comes back from the server, it will cause an error when we try to store the ACK. 

Note that this is unrelated to the bug reported in #7959 

### Proposed solution: 

Since we cant really prevent this scenario from happening, we should instead just account for it. So if we go to ACK an update for a channel that we no longer keep track of in the tower client, we should see this as a "Rogue" update and we should just at least keep track of the number of rogue updates that a session has backed up and we should not error in this case. 

## Affected versions

This would only have been possible since [0.16.1-beta](https://github.com/lightningnetwork/lnd/pull/7069)

**Comment by yyforyongyu** [comment]
> Basically if we have a updates queued in the tower client already but then the channel is closed,

Shouldn't we clean up the queue before marking it as closed in this case?

**Comment by ellemouton** [comment]
> Shouldn't we clean up the queue before marking it as closed in this case?

So we do actually check that the chan still exists when we pop from the queue but there are some other edge cases like: if we have already assigned a SeqNum to an update or we have already backed up the update and we get an ACK back from the server. 

We can eventually try to account for all these small edge cases but they will require quite a big re-design and migration since the channel ID is not available in an indexed way in the DB until the ACK is written. 

So I think for now, this "catch all" approach is fine due to this also just being a small edge case. Worst that can happen is that we backup a few updates for an already closed channel.
